mlflow/pyfunc/__init__.py

```diff
@@ -634,8 +663,15 @@ def load_model(
 
     _add_code_from_conf_to_system_path(local_path, conf, code_key=CODE)
     data_path = os.path.join(local_path, conf[DATA]) if (DATA in conf) else local_path
+    model_config = _get_overridden_pyfunc_model_config(
+        conf.get(MODEL_CONFIG, None), model_config, _logger
+    )
+
     try:
-        model_impl = importlib.import_module(conf[MAIN])._load_pyfunc(data_path)
+        if model_config:
+            model_impl = importlib.import_module(conf[MAIN])._load_pyfunc(data_path, model_config)
```

Comment:
This won't work for modules that '_load_pyfunc' doesn't contain 'model_config'. 
Instead, could we used **kwargs in def load_model? And kwargs passed here should override existing load_config saved with the Model.

Reply:
It will. Take a closer look at the `if` statement. The change is backward compatible as models without a `load_config` always load the older version of the signature. There is an assumption though that users using `load_config` when saving a model will also provide code that can handle it. Shall we add a validation to it? I don't think we do that today for existing requirements. (for instance, to check that there is a method `predict`, the method has `input` parameters, etc.). Regarding using kwargs... I would suggest to do not. That can mean trouble in the future if we plan to add new parameters that start to conflict with those from the user. I rather keep `load_config`.

Reply:
I think @serena-ruan 's point is that, if I pass `model_config` to `mlflow.pyfunc.load_model()` for a model that doesn't support `model_config`, I get a nasty "TypeError: _load_pyfunc() takes 1 positional argument but 2 were given`

We should handle that case, but I don't think `**kwargs` helps / is the right way to do it. Can we do an `inspect.signature()`?

Reply:
This is handled correctly by the behavior of `_get_overridden_pyfunc_model_config` and the  `if` check. @santiagxf can we add a test for this case? Otherwise, I think the implementation is great

Reply:
Just added a test case for the scenario @dbczumar!